src/WebServer_JSON.ino

```diff
@@ -3,7 +3,75 @@
 #include "src/Globals/Plugins.h"
 #include "StringProviderTypes.h"
 
+// ********************************************************************************
+// Web Interface get CSV value from task
+// ********************************************************************************
+void handle_csvval()
+{
+  String htmlData(25, '\0'); // Reserve for error message
+  const int printHeader = getFormItemInt(F("header"), 1);
+  bool printHeaderValid = true;
+  if (printHeader != 1 && printHeader != 0)
+  {
+    htmlData = F("ERROR: Header not valid!\n");
+    printHeaderValid = false;
+  }
+
+  const taskIndex_t taskNr    = getFormItemInt(F("tasknr"), INVALID_TASK_INDEX);
+  const bool taskValid = validTaskIndex(taskNr);
+  if (!taskValid)
+  {
+    htmlData = F("ERROR: TaskNr not valid!\n");
+  }
+
+  const int INVALID_VALUE_NUM = INVALID_TASKVAR_INDEX + 1;
+  const taskVarIndex_t valNr    = getFormItemInt(F("valnr"), INVALID_VALUE_NUM);
+  bool valueNumberValid = true;
+  if (valNr != INVALID_VALUE_NUM && !validTaskVarIndex(valNr))
+  {
+    htmlData = F("ERROR: ValueId not valid!\n");
+    valueNumberValid = false;
+  }
 
+  TXBuffer.startJsonStream();
+  if (taskValid && valueNumberValid && printHeaderValid)
+  {
+    const deviceIndex_t DeviceIndex = getDeviceIndex_from_TaskIndex(taskNr);
+
+    if (validDeviceIndex(DeviceIndex))
+    {
+      LoadTaskSettings(taskNr);
+      byte taskValCount = Device[DeviceIndex].ValueCount;
+      uint16_t stringReserveSize = (valNr == INVALID_VALUE_NUM ? 1 : taskValCount) * 24;
+      if (printHeader)
+      {
+        stringReserveSize += 11;
+      }
+      htmlData.reserve(stringReserveSize);
+
+      if (printHeader)
+      {
+        htmlData += "Name";
+        htmlData += ';';
+        htmlData += "Value";
+        htmlData += '\n';
+      }
+
+      for (byte x = 0; x < taskValCount; x++)
+      {
+        if (valNr == INVALID_VALUE_NUM || valNr == x)
+        {
+          htmlData += String(ExtraTaskSettings.TaskDeviceValueNames[x]);
+          htmlData += ';';
+          htmlData += formatUserVarNoCheck(taskNr, x);
+          htmlData += '\n';
```

Comment:
You still output the data in this form:

```
Name;Value
var1;1.234
var2;2.345
var3;3.456
```

CSV is more like this:

```
var1;var1;var3
1.234;2.345;3.456
```

This way you only need to send the 'header' once, or maybe even leave it out on a (next) request.

Reply:
Ahh, so:
Header(optional) -> var1;var1;var3
values                  ->1.234;2.345;3.456

Right?

Reply:
yes, but fought it is raw =)
maybe format is like:
`name=value`

Reply:
> yes, but fought it is raw =)
> maybe format it like:
> `name=value`

He just changed all into making it look like CSV (even the URL endpoint) :)

Reply:
Ok, Fixed ;)